common/lib/xmodule/xmodule/contentstore/content.py

```diff
@@ -137,32 +142,90 @@ def get_location_from_path(path):
                 return AssetKey.from_string(path[1:])
 
     @staticmethod
-    def convert_legacy_static_url_with_course_id(path, course_id):
+    def get_asset_key_from_path(course_key, path):
+        """
+        Parses a path, extracting an asset key or creating one.
+
+        Args:
+            course_key: key to the course which owns this asset
+            path: the path to said content
+
+        Returns:
+            AssetKey: the asset key that represents the path
+        """
+
+        # Clean up the path, removing any static prefix.
+        if path.startswith('/static/'):
+            path = path[len('/static/'):]
+
+        asset_key = None
+
+        # First, try to parse the path as is. This will catch any /c4x/*-style paths.
+        try:
+            asset_key = AssetKey.from_string(path)
+        except InvalidKeyError:
+            # Do we have a leading forward slash? Strip it and try again.
+            # This will catch asset key paths e.g. /asset-v1:.....
+            try:
+                asset_key = AssetKey.from_string(path[1:])
+            except InvalidKeyError:
+                pass
```

Comment:
Ah. If it's just copying existing logic, I wouldn't worry too much about it. In general though, please try to create the strings you need, rather than having offsets and indexing into them. It's just a bit clearer what's going on.